ilc/server/registry/Registry.js

```diff
@@ -99,10 +99,21 @@ module.exports = class Registry {
         return res.data;
     };
 
-    #getTemplate = async (templateName, locale) => {
+    #getTemplate = async (templateName, locale, domain) => {
         this.#logger.debug('Calling get template registry endpoint...');
-        const queryString = locale ? `?locale=${locale}` : '';
-        const tplUrl = urljoin(this.#address, 'api/v1/template', templateName, 'rendered') + queryString;
+
+        const params = new Map();
```

Comment:
What is the reason to use Map instead of simple Array?
Moreover - I don't see the necessity to use the intermediate variable, since regarding the specification of `URLSearchParams` - we can use `URLSearchParams.prototype.set` and `URLSearchParams.prototype.entries.length`

Reply:
Agree